spec/features/happy_journeys/js/when_choose_lead_mentor_course.rb

```diff
@@ -0,0 +1,193 @@
+      expect(page).to have_text("Get a Teacher Reference Number (TRN)")
+
+      page.click_link("Back")
+    end
+
+    expect_page_to_have(path: "/registration/teacher-reference-number", submit_form: true) do
+      page.choose("Yes", visible: :all)
+    end
+
+    expect(page).not_to have_content("Do you have a TRN?")
+
+    expect_page_to_have(path: "/registration/provider-check", submit_form: true) do
+      expect(page).to have_text("Have you already chosen an NPQ and provider?")
+      page.choose("Yes", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/teacher-catchment", axe_check: false, submit_form: true) do
+      page.choose("England", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/work-setting", submit_form: true) do
+      page.choose("Other", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/your-employment", submit_form: true) do
+      expect(page).to have_text("How are you employed?")
+      page.choose("As a lead mentor for an accredited ITT provider", visible: :all)
+    end
+
+    approved_itt_provider_legal_name = ::IttProvider.currently_approved.sample.legal_name
+
+    expect_page_to_have(path: "/registration/itt-provider", submit_form: true) do
+      expect(page).to have_text("Enter the name of the ITT provider you are working with")
+      page.fill_in("Enter the name of the ITT provider you are working with", with: approved_itt_provider_legal_name)
+      page.click_button("Continue")
+    end
+
+    expect_page_to_have(path: "/registration/choose-your-npq", submit_form: true) do
+      expect(page).to have_text("What are you applying for?")
+      page.choose("NPQ for Leading Teacher Development (NPQLTD)", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/possible-funding", submit_form: true) do
+      expect(page).to have_text("If your provider accepts your application, you’ll qualify for DfE funding")
+      expect(page).to have_text("From the information you have provided, DfE scholarship funding should be available for the NPQ for Leading Teacher Development (NPQLTD).")
+    end
+
+    expect_page_to_have(path: "/registration/choose-your-provider", submit_form: true) do
+      expect(page).to have_text("Select your provider")
+      page.choose("Church of England", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/share-provider", submit_form: true) do
+      expect(page).to have_text("Sharing your NPQ information")
+      page.check("Yes, I agree my information can be shared", visible: :all)
+    end
+
+    allow(ApplicationSubmissionJob).to receive(:perform_later).with(anything)
+
+    expect_page_to_have(path: "/registration/check-answers", submit_button_text: "Submit", submit_form: true) do
+      expect_check_answers_page_to_have_answers(
+        {
+
+          "Course" => "NPQ for Leading Teacher Development (NPQLTD)",
+          "Employer" => "",
```

Comment:
Do we want empty answers here? We should be omitting these answers if the user didn't get asked the question.